presto/presto.sh

```diff
@@ -12,75 +12,106 @@
-
-# Download and unpack Presto server
-wget https://repo1.maven.org/maven2/com/facebook/presto/presto-server/${PRESTO_VERSION}/presto-server-${PRESTO_VERSION}.tar.gz
-tar -zxvf presto-server-${PRESTO_VERSION}.tar.gz
-mkdir /var/presto
-mkdir /var/presto/data
-
-# Copy required Jars
-cp ${CONNECTOR_JAR} presto-server-${PRESTO_VERSION}/plugin/hive-hadoop2
-
-# Configure Presto
-mkdir presto-server-${PRESTO_VERSION}/etc
-mkdir presto-server-${PRESTO_VERSION}/etc/catalog
-
-cat > presto-server-${PRESTO_VERSION}/etc/node.properties <<EOF
+readonly ROLE="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
+readonly PRESTO_MASTER_FQDN="$(/usr/share/google/get_metadata_value attributes/dataproc-master)"
+readonly WORKER_COUNT=$(/usr/share/google/get_metadata_value attributes/dataproc-worker-count)
+readonly CONNECTOR_JAR="$(find /usr/lib/hadoop/lib -name 'gcs-connector-*.jar')"
+readonly PRESTO_VERSION='0.177'
+readonly HTTP_PORT='8080'
+readonly INIT_SCRIPT='/usr/lib/systemd/system/presto.service'
+PRESTO_JVM_MB=0;
+PRESTO_QUERY_NODE_MB=0;
+PRESTO_RESERVED_SYSTEM_MB=0;
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function get_presto(){
+  # Download and unpack Presto server
+  wget https://repo1.maven.org/maven2/com/facebook/presto/presto-server/${PRESTO_VERSION}/presto-server-${PRESTO_VERSION}.tar.gz
+  tar -zxvf presto-server-${PRESTO_VERSION}.tar.gz
+  mkdir -p /var/presto/data
+}
+
+function calculate_memory(){
+  # Compute memory settings based on Spark's settings.
+  # We use "tail -n 1" since overrides are applied just by order of appearance.
+  local spark_executor_mb;
+  spark_executor_mb=$(grep spark.executor.memory \
+    /etc/spark/conf/spark-defaults.conf \
+    | tail -n 1  \
+    | sed 's/.*[[:space:]=]\+\([[:digit:]]\+\).*/\1/')
+
+  local spark_executor_cores;
+  spark_executor_cores=$(grep spark.executor.cores \
+    /etc/spark/conf/spark-defaults.conf \
+    | tail -n 1 \
+    | sed 's/.*[[:space:]=]\+\([[:digit:]]\+\).*/\1/')
+
+  local spark_executor_overhead_mb;
+  if (grep spark.yarn.executor.memoryOverhead /etc/spark/conf/spark-defaults.conf); then
+    spark_executor_overhead_mb=$(grep spark.yarn.executor.memoryOverhead \
+      /etc/spark/conf/spark-defaults.conf \
+      | tail -n 1 \
+      | sed 's/.*[[:space:]=]\+\([[:digit:]]\+\).*/\1/')
+  else
+    # When spark.yarn.executor.memoryOverhead couldn't be found in
+    # spark-defaults.conf, use Spark default properties:
+    # executorMemory * 0.10, with minimum of 384
+    local min_executor_overhead=384
+    spark_executor_overhead_mb=$(( ${spark_executor_mb} / 10 ))
+    spark_executor_overhead_mb=$(( ${spark_executor_overhead_mb}>${min_executor_overhead}?${spark_executor_overhead_mb}:${min_executor_overhead} ))
+  fi
+  local spark_executor_count;
+  spark_executor_count=$(( $(nproc) / ${spark_executor_cores} ))
+
+  # Add up overhead and allocated executor MB for container size.
+  local spark_container_mb;
```

Comment:
Can all of these `local foo; foo = 'bar'` be declared and assigned in one line? (Or is that against the style guide?)

Reply:
Syntactically they can, but according to [shell style guide](https://google.github.io/styleguide/shell.xml?showone=Use_Local_Variables#Use_Local_Variables) they shouldn't:
> Declaration and assignment should be on different lines.